binfmt/binfmt_execmodule.c

```diff
@@ -150,14 +151,26 @@ int exec_module(FAR const struct binary_s *binp,
         }
     }
 
+  /* Make a copy of the environment here */
+
+  if (envp)
+    {
+      envp = binfmt_copyenv(envp);
```

Comment:
look like we should change to:
```
envp = binfmt_copyenv(envp ? envp : get_environ_ptr);
```
It's too late to handle NULL pointer in env_dup.

Reply:
Yes you are right, and it is better to do it here anyway as it keeps legacy apps that use binfmt_execmodule() working too (if any exist).